packages/project/src/db/project.ts

```diff
@@ -204,25 +205,40 @@ const clone = async (
       ? await db.build.loadByProjectId(project.id, "dev")
       : await db.build.loadByProjectId(project.id, "prod");
 
-  const projectId = uuid();
-  authorizeProject.registerProjectOwner({ projectId }, context);
+  const newProjectId = uuid();
+  await authorizeProject.registerProjectOwner(
+    { projectId: newProjectId },
+    context
+  );
 
   const clonedProject = await prisma.$transaction(async (client) => {
     const clonedProject = await client.project.create({
       data: {
-        id: projectId,
+        id: newProjectId,
         userId: userId,
         title: title ?? project.title,
         domain: generateDomain(project.title),
       },
     });
 
     await db.build.create(
-      { projectId: clonedProject.id, env: "dev", sourceBuild: build },
+      { projectId: newProjectId, env: "dev", sourceBuild: build },
       context,
       client
     );
 
+    await cloneAssets(
+      {
+        fromProjectId: project.id,
+        toProjectId: newProjectId,
+
+        // Permission check on newProjectId will fail until this transaction is committed.
```

Comment:
I would add comment like this into function definition too using /** **/ so it will be visible in vscode

Reply:
Added